src/test/scala/zio/rocksdb/RocksDBSpec.scala

```diff
@@ -2,78 +2,74 @@ package zio.rocksdb
-      suite("RocksDB")(
-        testM("get/put") {
-          val key   = "key".getBytes(UTF_8)
-          val value = "value".getBytes(UTF_8)
+object RocksDBSpec extends DefaultRunnableSpec {
+  override def spec = suite("RocksDB")(
+    testM("get/put") {
+      val key   = "key".getBytes(UTF_8)
+      val value = "value".getBytes(UTF_8)
 
-          val test =
-            for {
-              _      <- rocksdb.put(key, value)
-              result <- rocksdb.get(key)
-            } yield assert(result, isSome(equalTo(value)))
+      val test =
+        for {
+          _      <- rocksdb.put(key, value)
+          result <- rocksdb.get(key)
+        } yield assert(result)(isSome(equalTo(value)))
 
-          test.provideManaged(Utils.tempDB)
-        },
-        testM("delete") {
-          val key   = "key".getBytes(UTF_8)
-          val value = "value".getBytes(UTF_8)
+      test.provideLayer(database)
+    },
+    testM("delete") {
+      val key   = "key".getBytes(UTF_8)
+      val value = "value".getBytes(UTF_8)
 
-          val test =
-            for {
-              _      <- rocksdb.put(key, value)
-              before <- rocksdb.get(key)
-              _      <- rocksdb.delete(key)
-              after  <- rocksdb.get(key)
-            } yield assert(before, isSome(equalTo(value))) && assert(after, isNone)
+      val test =
+        for {
+          _      <- rocksdb.put(key, value)
+          before <- rocksdb.get(key)
+          _      <- rocksdb.delete(key)
+          after  <- rocksdb.get(key)
+        } yield assert(before)(isSome(equalTo(value))) && assert(after)(isNone)
 
-          test.provideManaged(Utils.tempDB)
-        },
-        testM("newIterator") {
-          val data = (1 to 10).map(i => (s"key$i", s"value$i")).toList
+      test.provideLayer(database)
+    },
+    testM("newIterator") {
+      val data = (1 to 10).map(i => (s"key$i", s"value$i")).toList
 
-          val test =
-            for {
-              _          <- RIO.foreach(data) { case (k, v) => rocksdb.put(k.getBytes(UTF_8), v.getBytes(UTF_8)) }
-              results    <- rocksdb.newIterator.runCollect
-              resultsStr = results.map { case (k, v) => new String(k, UTF_8) -> new String(v, UTF_8) }
-            } yield assert(resultsStr, hasSameElements(data))
+      val test =
+        for {
+          _          <- RIO.foreach(data) { case (k, v) => rocksdb.put(k.getBytes(UTF_8), v.getBytes(UTF_8)) }
+          results    <- rocksdb.newIterator.runCollect
+          resultsStr = results.map { case (k, v) => new String(k, UTF_8) -> new String(v, UTF_8) }
+        } yield assert(resultsStr)(hasSameElements(data))
 
-          test.provideManaged(Utils.tempDB)
-        }
-      )
-    )
-
-object Utils {
-  def tempDB: Managed[Throwable, RocksDB] =
-    tempDir.flatMap { dir =>
-      val opts = new Options().setCreateIfMissing(true)
-      RocksDB.Live.open(opts, dir.toAbsolutePath.toString)
+      test.provideLayer(database)
```

Comment:
It looks like you are providing the layer to each of your tests here. You could simplify a little by pulling this out and providing it to the whole suite using `Spec#provideCustomLayer`.

Reply:
Done.